src/main/scala/devices/debug/DebugTransport.scala

```diff
@@ -65,24 +65,29 @@ class DTMInfo extends Bundle {
 /** A wrapper around JTAG providing a reset signal and manufacturer id. */
 class SystemJTAGIO extends Bundle {
   val jtag = Flipped(new JTAGIO(hasTRSTn = false))
-  val reset = Input(Bool())
+  val reset = Input(Reset())
   val mfr_id = Input(UInt(11.W))
   val part_number = Input(UInt(16.W))
   val version = Input(UInt(4.W))
 }
 
 class DebugTransportModuleJTAG(debugAddrBits: Int, c: JtagDTMConfig)
-  (implicit val p: Parameters) extends Module  {
+  (implicit val p: Parameters) extends RawModule  {
 
   val io = IO(new Bundle {
+    val jtag_clock = Input(Clock())
+    val jtag_reset = Input(Reset())
+    val tapReset = Input(Bool())
```

Comment:
I am curious why we kept this one as `Bool`. I actually forget if there was something in the JTAG spec that implied something should be sync reset, or it was general laziness in the fact that we had to put down `AsyncResetRegVec` that kept much of the DTM as synchronous reset.

Reply:
I rearranged this and renamed it to `tapIsInTestLogicReset` to make it clearer that it's not a "reset".  JTAG spec says that the IDCODE IR is loaded by traversing the TAP through the Test-Logic-Reset state.  This is done synchronously to avoid creating an async reset within the design.